src/details/ArborX_DetailsDistributedTreeImpl.hpp

```diff
@@ -32,8 +32,25 @@
 namespace ArborX
 {
 
+struct PairIndexRank
```

Comment:
If we are elevating to a user-facing type then define it in its own header and fix your header includes in the test print helpers.